Source/Particles/ElementaryProcess/QEDPhotonEmission.H

```diff
@@ -0,0 +1,208 @@
+    {
+        using namespace amrex;
+
+        constexpr ParticleReal me = PhysConst::m_e;
+        constexpr ParticleReal one_over_me = 1._prt/me;
+
+        const ParticleReal w = src.m_rdata[PIdx::w][i_src];
+        const ParticleReal ux = src.m_rdata[PIdx::ux][i_src];
+        const ParticleReal uy = src.m_rdata[PIdx::uy][i_src];
+        const ParticleReal uz = src.m_rdata[PIdx::uz][i_src];
+        const ParticleReal ex = src.m_rdata[PIdx::Ex][i_src];
+        const ParticleReal ey = src.m_rdata[PIdx::Ey][i_src];
+        const ParticleReal ez = src.m_rdata[PIdx::Ez][i_src];
+        const ParticleReal bx = src.m_rdata[PIdx::Bx][i_src];
+        const ParticleReal by = src.m_rdata[PIdx::By][i_src];
+        const ParticleReal bz = src.m_rdata[PIdx::Bz][i_src];
+
+        // Particle momentum is stored as gamma * velocity.
+        // Convert to m * gamma * velocity before applying the emission functor.
+        auto px = ux*me;
+        auto py = uy*me;
+        auto pz = uz*me;
+
+        auto g_w = 0.0_rt;
+        auto g_px = 0.0_rt;
+        auto g_py = 0.0_rt;
+        auto g_pz = 0.0_rt;
+
+        m_emission_functor.operator()<1>(
+            &px, &py, &pz,
+            ex, ey, ez,
+            bx, by, bz,
+            w,
+            &g_px, &g_py, &g_pz,
+            &g_w);
+
+        // Then convert back to WarpX convention.
+        src.m_rdata[PIdx::ux][i_src] = px*one_over_me;
+        src.m_rdata[PIdx::uy][i_src] = py*one_over_me;
+        src.m_rdata[PIdx::uz][i_src] = pz*one_over_me;
+
+        dst.m_rdata[PIdx::w][i_dst] = g_w;
+        dst.m_rdata[PIdx::ux][i_dst] = g_px*one_over_me;
+        dst.m_rdata[PIdx::uy][i_dst] = g_py*one_over_me;
+        dst.m_rdata[PIdx::uz][i_dst] = g_pz*one_over_me;
+
+        //Initialize the optical depth component of the source species.
+        src.m_runtime_rdata[m_opt_depth_runtime_comp][i_src] =
+            m_opt_depth_functor();
+    }
+
+private:
+    const QuantumSynchrotronGetOpticalDepth
+        m_opt_depth_functor;  /*!< A copy of the functor to initialize the optical depth of the source species. */
+    const QuantumSynchrotronGeneratePhotonAndUpdateMomentum
+        m_emission_functor;  /*!< A copy of the functor to generate photons. It contains only pointers to the lookup tables.*/
+    const int m_opt_depth_runtime_comp = 0;  /*!< Index of the optical depth component of source species*/
+};
+
+
+/**
+* \brief Free function to call to remove immediately
+* low energy photons by setting their ID to -1.
+*
+* @tparam PTile particle tile type
+* @param[in,out] ptile a particle tile
+* @param[in] old_size the old number of particles
+* @param[in] num_added the number of photons added to the tile
+* @param[in] t_energy_threshold the energy threshold
+*/
+template <typename PTile>
+void cleanLowEnergyPhotons(
+PTile& ptile,
+const int old_size, const int num_added,
+const amrex::ParticleReal t_energy_threshold)
```

Comment:
```suggestion
    PTile& ptile,
    const int old_size, const int num_added,
    const amrex::ParticleReal t_energy_threshold)
```
or am I missing some indentation conventions?

Reply:
oh sorry! You are right!